### PR TITLE
[Improvement/feature] New Inspector popup UI 

### PR DESCRIPTION
**Pull request by arpitnath**
- [x] JSON Tree Viewer component (no lib dep)
- [x] new UI for the sidebar inspector popup
- [x] able to copy the entire collection to the clipboard with a single click
- [x] able to dispatch actions for selected queries: **RunQuery**
- [x] able to dispatch actions for selected components: delete a component
- [x] selecting a component (from the JSON tree) selects the component in the editor and 
opens widget inspector
- [x] selecting a widget in the editor selects and expands the node: widget from the json tree
- [x] Toggle Pin and unpin icons accordingly 
- [x] Dark theme UI

### CODE REVIEW

**Comment by Nandinisaha13** [comment]
<html>
<body>
<google-sheets-html-origin>Test Cases

Test Case | Steps | Result
-- | -- | --
UI for pin button. | Create and application. | Application gets created.
  | Click on inspector on left side bar. | The inspector opens up.
  | Observe. | UI of the pin button changed.
UI for unpin button. | Create and application. | Application gets created.
  | Click on inspector on left side bar. | The inspector opens up.
  | Observe. | UI of the pin button changed.
  | Click on the pin button. | UI of the unpin button changed.
UI in dark mode. | Create and application. | Application gets created.
  | Click on inspector on left side bar. | The inspector opens up.
  | Observe. | New UI is visible.
UI in light mode. | Create and application. | Application gets created.
  | Click on inspector on left side bar. | The inspector opens up.
  | Observe. | New UI is visible.
Able to copy the entire collection to the clipboard with a single click. | Create and application. | Application gets created.
  | Click on inspector on left side bar. | The inspector opens up.
  | Click on the 3 dots and select copy value. | The value is copied to the clipboard.
Able to copy the entire absolute path to the clipboard with a single click. | Create and application. | Application gets created.
  | Click on inspector on left side bar. | The inspector opens up.
  | Click on the copy icon. | The absolute value is copied to the clipboard.
Selecting a component (from the JSON tree) selects the component in the editor and opens widget inspector | Create and application. | Application gets created.
  | Click on inspector on left side bar. | The inspector opens up.
  | Click on the pin button. | It pins the inspector.
  | Add any widget on canvas. | The widget is added to the canvas.
  | Click on a component from the JSON tree. | The component should be selected and widget inspector opens.
Able to dispatch actions for selected queries: RunQuery | Create and application. | Application gets created.
  | Click on inspector on left side bar. | The inspector opens up.
  | Click on the pin button. | It pins the inspector.
  | Create a query. | The query gets created.
  | Run the query from inspector. | The query is executed.
Able to dispatch actions for selected components: delete a component | Create and application. | Application gets created.
  | Click on inspector on left side bar. | The inspector opens up.
  | Click on the pin button. | It pins the inspector.
  | Add any widget on canvas. | The widget is added to the canvas.
  | Click on delete icon on the inspector. | The widget is deleted from the canvas.
Selecting a widget in the editor selects and expands the node: widget from the json tree | Create and application. | Application gets created.
  | Click on inspector on left side bar. | The inspector opens up.
  | Click on the pin button. | It pins the inspector.
  | Add any widget on canvas. | The widget is added to the canvas.
  | Click on components on inspector. | The components should be opened.
  | Click on the widget on canvas. | The components in the inspector gets selected & opens up.


</body>
</html>

File: frontend/src/_ui/JSONTreeViewer/JSONTreeViewer.jsx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+  getLength(type, collection) {
+    if (!collection) return 0;
+    if (type === 'Object') {
+      return Object.keys(collection).length;
+    } else if (type === 'Array') {
+      return collection.length;
+    }
+
+    return Infinity;
```
Comment (gondar00) [inline frontend/src/_ui/JSONTreeViewer/JSONTreeViewer.jsx:82]: probably return zero here?

File: frontend/src/_ui/JSONTreeViewer/JSONTreeViewer.jsx [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+          map.set(newPath, { type: _type });
+          buildMap(value, newPath);
+        } else if (_.isArray(value)) {
+          map.set(newPath, { type: _type });
+          buildMap(value, newPath);
+        } else if (_.isFunction(value)) {
+          map.set(newPath, { type: _type });
+        } else {
+          map.set(newPath, { type: _type });
+        }
```
Comment (gondar00) [inline frontend/src/_ui/JSONTreeViewer/JSONTreeViewer.jsx:176]:
```suggestion
        map.set(newPath, { type: _type });
        if (_.isObject(value) || _.isArray(value)) {
          buildMap(value, newPath);
        }
```

**Comment by arpitnath** [comment]
@gondar00 new app link with all the changes resolved: https://inspector-uijson.herokuapp.com/

**Review by gondar00 — APPROVED** [review]

**Review by sherfin94 — APPROVED** [review]
Generally looks fantastic 🔥  , one small improvement could be to have a very mild highlighting on hover:
image
@arpitnath

**Pull request merged**